### DISCUSSION TITLE
Use sparse bundle disk image for macOS vms

### DISCUSSION THREAD

**Discussion by matthewruzzi** [utmapp/UTM#4436] [Ideas]
Currently, UTM (and parallels) create a full sized disk image for the capacity chosen when setting it up. This makes it difficult to back up because the entire vm and all unused space in it will have to be backed up every time a small change is made. Would it be possible to use a sparce bundle disk image that only grows when space inside of it is used? For linux and windows vms, Parallels desktop uses a .hdd file that is a bundle of files that grow as space is used. Incrementally backing a vm up with rsync is a lot faster because only the changes have to be copied because a new file in the bundle is created every time a snapshot is taken and old files are not modified.

**Comment by stendro** [comment]
I'm using rsyncosx, and I have the problem that the *.utm file remains the same on the outside, only if you open it you can see the qcow file has changed. Therefore rsync can't work because it always thinks the VM is the same.
